src/QuadForm/Torsion.jl

```diff
@@ -733,17 +759,383 @@ function preimage(f::TorQuadModMor, a::TorQuadModElem)
+GrpAb: (Z/3)^5
+Gram matrix of the quadratic form with values in Q/2Z
+[2//3   2//3      0      0      0]
+[2//3   2//3   2//3      0   2//3]
+[   0   2//3   2//3   2//3      0]
+[   0      0   2//3   2//3      0]
+[   0   2//3      0      0   2//3]
+
+julia> U = torsion_quadratic_module(QQ[4//3    0    0    0    0;
+                                          0 4//3    0    0    0;
+                                          0    0 4//3    0    0;
+                                          0    0    0 4//3    0;
+                                          0    0    0    0 4//3])
+Finite quadratic module over Integer Ring with underlying abelian group
+GrpAb: (Z/3)^5
+Gram matrix of the quadratic form with values in Q/2Z
+[4//3      0      0      0      0]
+[   0   4//3      0      0      0]
+[   0      0   4//3      0      0]
+[   0      0      0   4//3      0]
+[   0      0      0      0   4//3]
+
+julia> bool, phi = is_isometric_with_map(T,U)
+(true, Map with following data
+Domain:
+=======
+TorQuadMod [2//3 2//3 0 0 0; 2//3 2//3 2//3 0 2//3; 0 2//3 2//3 2//3 0; 0 0 2//3 2//3 0; 0 2//3 0 0 2//3]
+Codomain:
+=========
+TorQuadMod [4//3 0 0 0 0; 0 4//3 0 0 0; 0 0 4//3 0 0; 0 0 0 4//3 0; 0 0 0 0 4//3])
+
+julia> is_bijective(phi)
+true
+
+julia> T2, _ = sub(T, [-gens(T)[4], gens(T)[2]+gens(T)[3]+gens(T)[5]])
+(TorQuadMod: (Z/3)^2 [2//3 1//3; 1//3 2//3], Map with following data
+Domain:
+=======
+TorQuadMod [2//3 1//3; 1//3 2//3]
+Codomain:
+=========
+TorQuadMod [2//3 2//3 0 0 0; 2//3 2//3 2//3 0 2//3; 0 2//3 2//3 2//3 0; 0 0 2//3 2//3 0; 0 2//3 0 0 2//3])
+
+julia> U2, _ = sub(T, [gens(T)[4], gens(T)[2]+gens(T)[3]+gens(T)[5]])
+(TorQuadMod: (Z/3)^2 [2//3 2//3; 2//3 2//3], Map with following data
+Domain:
+=======
+TorQuadMod [2//3 2//3; 2//3 2//3]
+Codomain:
+=========
+TorQuadMod [2//3 2//3 0 0 0; 2//3 2//3 2//3 0 2//3; 0 2//3 2//3 2//3 0; 0 0 2//3 2//3 0; 0 2//3 0 0 2//3])
+
+julia> bool, phi = is_isometric_with_map(U2, T2)
+(true, Map with following data
+Domain:
+=======
+TorQuadMod [2//3 2//3; 2//3 2//3]
+Codomain:
+=========
+TorQuadMod [2//3 1//3; 1//3 2//3])
+
+julia> is_bijective(phi)
+true
+```
+"""
+function is_isometric_with_map(T::TorQuadMod, U::TorQuadMod)
+  if T === U
+    return (true, id_hom(T))
+  end
+  # the zero map for default output
+  hz = hom(T, U, zero_matrix(ZZ, ngens(T), ngens(U)))
+  if order(T) != order(U)
+    return (false, hz)
+  end
+  @req (modulus(T) == 1 && modulus(U) == 1) "Only implemented for torsion quadratic module with bilinear modulus 1"
+  if elementary_divisors(T) != elementary_divisors(U)
+    return (false, hz)
+  end
+  if is_degenerate(T) != is_degenerate(U)
+    return (false, hz)
+  end
+  # if they have no elementary divisors, then they are trivial and therefore isometric
+  if length(elementary_divisors(T)) == 0
+    return (true, hz)
+  end
+  # they should have the same parity
+  if modulus_quadratic_form(T) != modulus_quadratic_form(U)
+    return (false, hz)
+  end
+  if !is_degenerate(T)
```

Comment:
Does test semiregularity. Maybe add a function `is_semi_regular`?
```suggestion
  if 1==(order(radical_quadratic(T)[1]))
```

Reply:
For abelian groups there is `is_trivial`.